.github/workflows/publish-packages.yml

```diff
@@ -0,0 +1,89 @@
+name: Publish packages
+
+on:
+  push:
+    # possibly could be improved with a proper regex
+    # ex: https://stackoverflow.com/a/58869470
+    tags:
+      - v*.*.*
+
+jobs:
+  setup:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up Homebrew
+        id: set-up-homebrew
+        uses: Homebrew/actions/setup-homebrew@master
+
+      - name: Install moreutils
+        run: brew install moreutils
+
+      - name: Install git-toolbelt
+        run:
+          brew install fzf && brew tap nvie/tap && brew install
+          nvie/tap/git-toolbelt
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.base_ref }}
+
+      - name: Set git tag
+        id: git_tag
+        run: echo "value=${GITHUB_REF#refs/*/}" >> $GITHUB_OUTPUT
+
+      - name: Set release version
+        id: version
+        env:
+          GIT_TAG: ${{ steps.git_tag.outputs.value }}
+        run: echo "value=${GIT_TAG#v}" >> $GITHUB_OUTPUT
+
+      - name: Set npm tag
+        id: npm_tag
+        env:
+          GIT_TAG: ${{ steps.git_tag.outputs.value }}
+        run: |
+          get_tag_if_exists () {
+            if grep -q "-" <<< "$1"; then
+                echo "${1##*-}" | sed 's/[0-9]//g'
+            else 
+                echo "latest"
+            fi
+          }
+          echo value=$(get_tag_if_exists "$GIT_TAG") >> $GITHUB_OUTPUT
+
+      - name: Check values
+        run: |
+          echo ${{steps.git_tag.outputs.value}}
+          echo ${{steps.version.outputs.value}}
+          echo ${{steps.npm_tag.outputs.value}}
+
+      - name: Setup git config
+        run: |
+          git config user.name "GitHub Actions Bot"
+          git config user.email "<>"
+
+      - name: Setup Node.js environment
+        uses: actions/setup-node@v3.6.0
+        with:
+          node-version: 16.17.1
+
+      - name: Preparing release
+        env:
+          VERSION: ${{steps.version.outputs.value}}
+        run: ./scripts/release.sh -V "$VERSION"
```

Comment:
Could we flip these sections, so that the "setup git" and "prepare release" are closer together?

```suggestion
      - name: Setup Node.js environment
        uses: actions/setup-node@v3.6.0
        with:
          node-version: 16.17.1

      - name: Setup git config
        run: |
          git config user.name "GitHub Actions Bot"
          git config user.email "<>"

      - name: Preparing release
        env:
          VERSION: ${{steps.version.outputs.value}}
        run: ./scripts/release.sh -V "$VERSION"
```

Here, I'm assuming that the "setup git" part is required because the "preparing release" section will create a new git commit, right?

Reply:
That's indeed the case, the created tag within the workflow did not trigger another github action 🎉 
![Screen Shot 2023-02-23 at 2 03 25 PM](https://user-images.githubusercontent.com/6947265/221005347-0e9ae3cf-761c-40d7-ab15-0eeefdf5e13f.png)
![Screen Shot 2023-02-23 at 2 03 19 PM](https://user-images.githubusercontent.com/6947265/221005351-7259649b-6607-4047-8229-1ff230c5d95c.png)

Reply:
What happens to your local tag in that case? Will this conflict be resolved when you do `git pull` next?

Reply:
It will conflict on the git pull, with vscode you get a popup that solves the conflict for you
<img width="361" alt="Screen Shot 2023-02-23 at 3 21 53 PM" src="https://user-images.githubusercontent.com/6947265/221021415-451717a5-fc0a-436f-a376-e44b641d9152.png">

Reply:
Perhaps some kind of `temp-v1.0.1-test1` tag could be created with `npm publish v1.0.1-test1` so we wouldn't have to delete it, and CI creates a proper one. 
That would mean that we'd get double the tags though, not sure we necessarily want that.

Reply:
Alternatively, you can change the local script to delete the tag directly after pushing it out?

```
git tag "$tag" && git push origin "$tag" && git tag -d "$tag"
```

Reply:
Well, there we go simple and effective, thank you!